File: Serene/Serene.Script/Membership/ForgotPasswordPanel.cs

namespace Serene.Membership
{
    using jQueryApi;
    using Serenity;
    using System.Html;

    [Panel, FormKey("Membership.ForgotPassword")]
    public class ForgotPasswordPanel : PropertyPanel<ForgotPasswordRequest>
    {
        public ForgotPasswordPanel(jQueryObject container)
            : base(container)
        {
            this.ById("SubmitButton").Click((s, e) =>
            {
                e.PreventDefault();

                if (!ValidateForm())
                    return;

                var request = GetSaveEntity();
                Q.ServiceCall(new ServiceCallOptions
                {
                    Url = Q.ResolveUrl("~/Account/ForgotPassword"),
                    Request = request,
                    OnSuccess = response =>
                    {
                        Q.Information(Q.Text("Forms.Membership.ForgotPassword.Success"), () =>
                        {
                            Window.Location.Href = Q.ResolveUrl("~/");
                        }, new ConfirmOptions());
                    }
                });
            });
        }
    }
}

File: Serene/Serene.Core/Modules/Membership/Account/ForgotPassword/ForgotPasswordPanel.ts
namespace Serene.Membership {

    @Serenity.Decorators.registerClass()
    export class ForgotPasswordPanel extends Serenity.PropertyPanel<ForgotPasswordRequest, any> {

        protected getFormKey() { return ForgotPasswordForm.formKey; }

        private form: ForgotPasswordForm;

        constructor(container: JQuery) {
            super(container);

            this.form = new ForgotPasswordForm(this.idPrefix);

            this.byId('SubmitButton').click(e => {
                e.preventDefault();

                if (!this.validateForm()) {
                    return;
                }

                var request = this.getSaveEntity();
                Q.serviceCall({
                    url: Q.resolveUrl('~/Account/ForgotPassword'),
                    request: request,
                    onSuccess: response => {
                        Q.information(Q.text('Forms.Membership.ForgotPassword.Success'), () => {
                            window.location.href = Q.resolveUrl('~/');
                        });
                    }
                });
            });
        }
    }
}